### DISCUSSION TITLE
Solar Power

### DISCUSSION THREAD

**Discussion by Splodgeink** [David00/rpi-power-monitor#95] [General]
Hi,

I've followed the instructions as detailed in the wiki, but my dashboard is not showing any generation. The CT (ct2 in my case) is showing current under generation. I have been unable to find any clues in the code or docs; the Database info seems to be missing.

Any help gratefully received!

Cheers
Andy

**Comment by David00** [comment]
Hi @Splodgeink, are you trying out the new v0.3.0 beta, or is this for the version 0.2.0?  

In v0.2.0, you need to edit both the `power_monitor.py` and `config.py` files to get solar working.
[v0.2.0 docs](https://github.com/david00/rpi-power-monitor/wiki/Solar-Integration#step-2-minor-code-adjustments)

In v0.3.0, you just need to set the `type = production` for the CT in `config.toml`. 
[v0.3.0 docs](https://david00.github.io/rpi-power-monitor/docs/v0.3.0-beta/configuration#setting-current-transformer-ct-sensor-details)

**Comment by Splodgeink** [comment]
Hi David,
I'm using 0.2.0 and I've amended power_monitor.py as detailed in the wiki. I've also looked through config.py and, apart from calibrations, I cant see what needs changing. There's nothing in the solar power wiki that I can see.  (It's probably me!)

I didn't even see 0.3.0!

Andy

**Reply by David00** [reply]
Excellent - in v0.2.0, you have to do a `pip install .` after changing any of the config settings or `power_monitor.py` code.  I suspect that is the issue you're facing.

This is fixed in v0.3.0, along with several other improvements, and I would highly recommend updating to v0.3.0.  You can upgrade straight from v0.2.0 by following the steps on this page:
https://github.com/David00/rpi-power-monitor/releases/tag/v0.3.0-beta

**Comment by Splodgeink** [comment]
Yeah, I've done the "pip install ." many times!
My next step is to upgrade to 0.3.0, now I know it exists! Thanks for the links and hard work. FWIW I can see positive generation of over 20W  when looking at raw ct data.
Andy

**Reply by David00** [reply]
Awesome - let me know if you need a hand with the upgrade!

**Reply by Splodgeink** [reply]
0.3.0 installed and working. Solar production is showing -ve although the raw CT data show +ve. I need to get my head around the JSON stuff!

**Reply by Splodgeink** [reply]
So changing the setting in config.toml to true works, but then the output in Grafana doesnt. Where can I sort this as I like my solar production to appear as a positive value?
Cheers
Andy

**Reply by David00** [reply]
In the dashboard, the solar production panel is a combination of all CT channels that have a `type = production`.

I would double check that the correct CTs have a type of production in your config.toml.  Then, you can validate the raw readings by starting the power monitor in [terminal mode](https://david00.github.io/rpi-power-monitor/docs/v0.3.0-beta/advanced-usage.html#--mode) to make sure the CT is reporting positive values. (Or, you can check the graph in the Individual CT Data dashboard).

Also, double check that your Solar Power graph in Grafana does not have a `* -1` in the panel's configuration.  The latest version of the dashboard does not have it, but if you imported the JSON dashboard a while ago, you might have the `* -1` in the Solar Power panel configuration.  If you're not sure, you can just delete the Home Power Dashboard and re-import it using the JSON link here:
https://github.com/David00/rpi-power-monitor/wiki/Software-3.-Building-your-Dashboard#dashboard-options

**Reply by Splodgeink** [reply]
I cant delete a 'provisioned' dashboard?

Also when looking at the raw signals, I cant see the flow going negative on my mains incomer.
![Raw CT data](https://user-images.githubusercontent.com/53603374/227710716-71dfbd48-1fed-440b-8591-ab19958d8f87.JPG)

**Comment by Splodgeink** [comment]
![Home Dash](https://user-images.githubusercontent.com/53603374/227710731-afa7d8bb-2ed2-43d9-bfce-d2e53c492214.JPG)

**Reply by David00** [reply]
Something's not quite right - the home power usage chart should not be going into the negatives. Let's double check that you're running the most up to date version of 0.3.0 with this:

```
cd ~/rpi_power_monitor
git pull
git status
```
The `git status` command should show the following:

```
$ git status
On branch develop-v0.3.0
Your branch is up to date with 'origin/develop-v0.3.0'.
```

Have you confirmed that your non-production CTs are reporting positive power when there is no power production happening? Also, check that your solar CT is also reporting positive power.  You should be checking this in terminal mode or the individual CT data dashboard.

**Reply by Splodgeink** [reply]
I've done that:
![image](https://user-images.githubusercontent.com/53603374/231120497-c20d7dee-e901-4b90-813d-0613d0b41fa9.png)

**Comment by Splodgeink** [comment]
Also, I can't find a *-1 in the JSON

**Reply by David00** [reply]
It would be a lot easier to check the Grafana panel configuration in Grafana. Edit the solar panel graph:
![image](https://user-images.githubusercontent.com/17997328/227734270-b0949232-463d-44f7-8fa5-e5f286f9e32c.png)

and make sure there's no `* -1` operation listed in the query settings:

![image](https://user-images.githubusercontent.com/17997328/227734286-3ba0d0d0-dccb-48a1-a803-556a013e00fb.png)

**Reply by Splodgeink** [reply]
Here is what mine says:

![image](https://user-images.githubusercontent.com/53603374/231118467-a87a6236-bffe-4e22-a69f-28666e56837c.png)

**Reply by Splodgeink** [reply]
All CTs showing positive. CT2 is my Solar output.
![image](https://user-images.githubusercontent.com/53603374/231118916-d9b04576-8f48-4ad9-8686-7a3bb22975b0.png)

**Reply by Splodgeink** [reply]
config.toml extract:
![image](https://user-images.githubusercontent.com/53603374/231119883-8940b9b1-d33b-4285-aa41-bb2b2d048046.png)

**Reply by David00** [reply]
Ok - for your channel 2, you can try setting `reversed = true`. 

For your channel 1, if it is on your mains, the type should be `mains`.  Note that the `consumption` type is reserved for CTs measuring individual circuits or appliances.

**Comment by Splodgeink** [comment]
Sorry for the delay. Have implemented those suggestions and the readings are making much more sense, thank you.
Now to install the kiln monitor circuit!
